### PR TITLE
Anmandan/tenant policies

### PR DESCRIPTION
**Pull request by anmandan**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [ ] new API version 
      - [x] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [x] No, no need to refresh for updates in this PR

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [ ] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [ ] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://msazure.visualstudio.com/One/_workitems/create/Scenario?templateId=d1c215ec-5d6a-4cb3-a193-95ddd0a1ba96&ownerId=85bb92fb-21fe-41ca-9309-47f2c5f57515). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Review by xseeseesee — APPROVED** [review]

**Comment by xccc-msft** [comment]
@anmandan Please fix the CI errors before merging this PR. Thanks.

File: specification/subscription/resource-manager/Microsoft.Subscription/preview/2019-10-01-preview/subscriptions.json [hunk 1]
```diff
@@ -532,6 +532,72 @@
           }
         ]
       }
+    },
+    "/providers/Microsoft.Subscription/policies/default": {
```
Comment (majastrz) [inline specification/subscription/resource-manager/Microsoft.Subscription/preview/2019-10-01-preview/subscriptions.json:536]: >providers [](start = 6, length = 9)

You should include a list call for the singleton resource as well.

**Comment by majastrz** [comment]
>     "version": "2019-10-01-preview",

Preview API versions older than a year are considered to be stable API versions for the purposes of the versioning policy and as such you should not be modified. The open API bot has added a comment with an explanation why. The full versioning guidance is available at [https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md#versioning](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md#versioning).

You should introduce these APIs in a new API version.

---
Refers to: specification/subscription/resource-manager/Microsoft.Subscription/preview/2019-10-01-preview/subscriptions.json:4 in 22753a3. [](commit_id = 22753a355f9943d6a8b6615bc3bcc3b451a6c37f, deletion_comment = False)

File: specification/subscription/resource-manager/Microsoft.Subscription/preview/2019-10-01-preview/examples/changeTenantPolicy.json [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        "9792da87-c97b-410d-a97d-27021ba09ce6"
+      ]
+    }
+  },
+  "responses": {
+    "200": {
+      "body": {
+        "id": "291bba3f-e0a5-47bc-a099-3bdcb2a50a05",
+        "name": "Default",
+        "type": "Microsoft.Subscription/Policies/default.",
```
Comment (majastrz) [inline specification/subscription/resource-manager/Microsoft.Subscription/preview/2019-10-01-preview/examples/changeTenantPolicy.json:18]: >. [](start = 56, length = 1)

The type should be `Microsoft.Subscription/policies`

File: specification/subscription/resource-manager/Microsoft.Subscription/preview/2019-10-01-preview/examples/changeTenantPolicy.json [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+        "e879cf0f-2b4d-5431-109a-f72fc9868693",
+        "9792da87-c97b-410d-a97d-27021ba09ce6"
+      ]
+    }
+  },
+  "responses": {
+    "200": {
+      "body": {
+        "id": "291bba3f-e0a5-47bc-a099-3bdcb2a50a05",
+        "name": "Default",
```
Comment (majastrz) [inline specification/subscription/resource-manager/Microsoft.Subscription/preview/2019-10-01-preview/examples/changeTenantPolicy.json:17]: >Default [](start = 17, length = 7)

Should be `default` (lower case)

File: specification/subscription/resource-manager/Microsoft.Subscription/preview/2019-10-01-preview/examples/changeTenantPolicy.json [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+      "exemptedPrincipals": [
+        "e879cf0f-2b4d-5431-109a-f72fc9868693",
+        "9792da87-c97b-410d-a97d-27021ba09ce6"
+      ]
+    }
+  },
+  "responses": {
+    "200": {
+      "body": {
+        "id": "291bba3f-e0a5-47bc-a099-3bdcb2a50a05",
```
Comment (majastrz) [inline specification/subscription/resource-manager/Microsoft.Subscription/preview/2019-10-01-preview/examples/changeTenantPolicy.json:16]: >291bba3f-e0a5-47bc-a099-3bdcb2a50a05 [](start = 15, length = 36)

Given your current path, the `id` should be `/providers/Microsoft.Subscription/policies/default`. If you are intending to support more than one policy, then the paths would have to be adjusted to support it.

File: specification/subscription/resource-manager/Microsoft.Subscription/preview/2019-10-01-preview/examples/changeTenantPolicy.json (hunk 2 above)
Comment (majastrz) [reply specification/subscription/resource-manager/Microsoft.Subscription/preview/2019-10-01-preview/examples/changeTenantPolicy.json:18]: The comments in this file apply to all the examples.

---
In reply to: [553070641](https://github.com/Azure/azure-rest-api-specs/pull/11672#discussion_r553070641) [](ancestors = 553070641)

**Comment by majastrz** [comment]
Added some comments. Please take a look.

File: specification/subscription/resource-manager/Microsoft.Subscription/preview/2019-10-01-preview/subscriptions.json [hunk 5]
```diff
@@ -868,6 +964,71 @@
           "readOnly": true
         }
       }
+    },
+    "TenantPolicy": {
+      "description": "Tenant policy.",
+      "properties": {
+        "blockSubscriptionsLeavingTenant": {
```
Comment (wilcobmsft) [inline specification/subscription/resource-manager/Microsoft.Subscription/preview/2019-10-01-preview/subscriptions.json:971]: thoughts on just having a single enum property? E.g.:
```
"subscriptionTransfers": "NotAllowed | Allowed | IncomingOnly | OutgoingOnly"
```

File: specification/subscription/resource-manager/Microsoft.Subscription/preview/2019-10-01-preview/subscriptions.json [hunk 6]
```diff
@@ -532,6 +532,102 @@
           }
         ]
       }
+    },
+    "/providers/Microsoft.Subscription/policies/default": {
```
Comment (wilcobmsft) [inline specification/subscription/resource-manager/Microsoft.Subscription/preview/2019-10-01-preview/subscriptions.json:536]: Should we call it "tenantPolicies"? That way we can reserve the name "policies" for policies scoped to a subscription, should we ever have a scenario for that.

Comment (majastrz) [reply specification/subscription/resource-manager/Microsoft.Subscription/preview/2019-10-01-preview/subscriptions.json:536]: Makes sense to me.

---
In reply to: [554133578](https://github.com/Azure/azure-rest-api-specs/pull/11672#discussion_r554133578) [](ancestors = 554133578)

File: specification/subscription/resource-manager/Microsoft.Subscription/preview/2019-10-01-preview/subscriptions.json (hunk 5 above)
Comment (majastrz) [reply specification/subscription/resource-manager/Microsoft.Subscription/preview/2019-10-01-preview/subscriptions.json:971]: If you go down this path, make sure each value of the enum has a description that explains what it means (specified via the x-ms-enum extension)

---
In reply to: [554128860](https://github.com/Azure/azure-rest-api-specs/pull/11672#discussion_r554128860) [](ancestors = 554128860)

File: specification/subscription/resource-manager/Microsoft.Subscription/preview/2019-10-01-preview/subscriptions.json (hunk 6 above)
Comment (anmandan) [reply specification/subscription/resource-manager/Microsoft.Subscription/preview/2019-10-01-preview/subscriptions.json:536]: we have billingPolicies  which would be 
/providers/Microsoft.Billing/billingAccounts/{billingAccountId}/providers/Microsoft.Subscription/Policies/default

File: specification/subscription/resource-manager/Microsoft.Subscription/preview/2019-10-01-preview/subscriptions.json (hunk 5 above)
Comment (anmandan) [reply specification/subscription/resource-manager/Microsoft.Subscription/preview/2019-10-01-preview/subscriptions.json:971]: Our backend has boolean logic setup for this, which would need additional conversions

**Pull request closed without merging**